RockPaperScissors.js

```diff
@@ -0,0 +1,95 @@
+var gamesWon = 0;
+var gamesPlayed = 0;
+var userName = " ";
+var gameState = "nameInput";
+
+var main = function (input) {
+  // Check Game Mode
+  if (gameState === "nameInput") {
+    userName = input;
+    gameState = "gameTime";
+    return "gm, " + userName + "! Welcome to the game!";
+  } else {
+    //Validate input is string
+    if (typeof input === "string") {
+      if (
+        input.toLowerCase() === "rock" ||
+        input.toLowerCase() === "paper" ||
+        input.toLowerCase() === "scissors"
+      ) {
+        var outputValue = winOrLose(input.toLowerCase());
+        console.log(outputValue);
+        return outputValue;
+      } else {
+        return "Please input only rock, paper or scissors.";
+      }
+    }
+  }
+  var ouputValue = winOrLose(input);
+  return ouputValue;
+};
+
+//Function for the actual game
+var winOrLose = function (userAction) {
+  var cpuAction = computerAction();
+  var myOutputValue = " ";
+  gamesPlayed++;
+  //User selects rock
+  if (userAction === "rock") {
+    if (cpuAction === "rock") {
+      // Computer selects rock
+      myOutputValue = "The computer played " + cpuAction + ". It's a tie!";
+    } else if (cpuAction === "scissors") {
+      // Computer selects scissors
+      myOutputValue = "The computer played " + cpuAction + ". You won!";
+      gamesWon++;
+    } else if (cpuAction === "paper") {
+      // Computer selects paper
+      myOutputValue = "The computer played " + cpuAction + ". You lost!";
+    }
+  } else if (userAction === "paper") {
+    //User selects paper
+    if (cpuAction === "rock") {
+      // Computer selects rock
+      myOutputValue = "The computer played " + cpuAction + ". You won!";
+      gamesWon++;
+    } else if (cpuAction === "scissors") {
+      // Computer selects scissors
+      myOutputValue = "The computer played " + cpuAction + ". You lost!";
+    } else if (cpuAction === "paper") {
+      // Computer selects paper
+      myOutputValue = "The computer played " + cpuAction + ". It's a tie!";
+    }
```

Comment:
do we need to check all 3 conditions? since you know that cpuAction only returns 3 values, maybe the last outcome can just be a draw?